src/main/java/net/snowflake/client/core/arrow/DateConverter.java

```diff
@@ -86,7 +98,7 @@ public BigDecimal toBigDecimal(int index) {
 
   @Override
   public Timestamp toTimestamp(int index, TimeZone tz) throws SFException {
-    Date date = toDate(index, tz, true);
+    Date date = toDate(index, tz, this.useDateFormat);
```

Comment:
when we use the field here instead of the const it could cause the breaking change for the users - I think that we should minimize the impact on the users and add the session property  to keep the previous behaviour for the users in the changed methods